crates/precompile/src/bls12_381/blst.rs

```diff
@@ -314,3 +322,283 @@ pub(super) fn pairing_check(pairs: &[(blst_p1_affine, blst_p2_affine)]) -> bool
     // Check if the result is one (identity element)
     is_fp12_one(&final_result)
 }
+
+/// Encodes a G1 point in affine format into byte slice with padded elements.
+pub(super) fn encode_g1_point(input: *const blst_p1_affine) -> [u8; PADDED_G1_LENGTH] {
+    let mut out = [0u8; PADDED_G1_LENGTH];
+    // SAFETY: Out comes from fixed length array, input is a blst value.
+    unsafe {
+        fp_to_bytes(&mut out[..PADDED_FP_LENGTH], &(*input).x);
+        fp_to_bytes(&mut out[PADDED_FP_LENGTH..], &(*input).y);
+    }
+    out
+}
+
+/// Encodes a single finite field element into byte slice with padding.
+fn fp_to_bytes(out: &mut [u8], input: *const blst_fp) {
```

Comment:
can these functions take in references instead of raw pointers?

Reply:
Yep -- I'll remove raw pointers from the public and private API of the blst wrapper